bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -25,215 +31,375 @@
+    }
+
+    /**
+     * Remove a listener and don't get notified of new/removed brokers anymore.
+     *
+     * @param observer The observer
+     */
+    public void removeBrokersListener(MqttBrokersObserver observer) {
+        brokersObservers.remove(observer);
+    }
+
+    /**
+     * Return true if a broker listener has been added via addBrokersListener().
+     */
+    public boolean isBrokerObservers() {
+        return !brokersObservers.isEmpty();
+    }
+
+    /**
+     * Lookup an broker connection by name.
+     *
+     * @param brokerName to look for.
+     * @return existing connection or null
+     */
+    public MqttBrokerConnection getBrokerConnection(String brokerName) {
+        synchronized (brokerConnections) {
+            return brokerConnections.get(brokerName.toLowerCase());
+        }
+    }
+
+    /**
+     * Adds a broker to the service. The broker connection will not be altered (started/stopped),
+     * by adding it to the service.
+     *
+     * The broker connection will be identified by its name. The name must be unique within the service.
+     *
+     * @param connection The broker connection object
+     * @return Return true if the connection could be added successfully, return false if there is already
+     *         an existing connection with the same name.
+     */
+    public boolean addBrokerConnection(MqttBrokerConnection connection) {
+        synchronized (brokerConnections) {
+            final String brokerID = connection.getName().toLowerCase();
+            if (brokerConnections.containsKey(brokerID)) {
+                return false;
+            }
+            brokerConnections.put(brokerID, connection);
+            for (MqttBrokersObserver o : brokersObservers) {
+                o.brokerAdded(connection);
             }
         }
+        return true;
     }
 
-    private void stopAllBrokerConnections() {
-        /*
-         * No need to synchronize the access to the broker connections.
-         * We don't add or remove entries only use the existing ones and the map is a concurrent one.
-         */
-        for (final MqttBrokerConnection conn : brokerConnections.values()) {
-            logger.info("Stopping broker connection '{}'", conn.getName());
-            conn.close();
+    /**
+     * Add a broker by a configuration key-value map. You need to provide at least a "name" and an "url".
+     * Additional properties are "user","pwd","qos","retain","lwt","keepAlive","clientId", please read the
+     * service configuration documentation for a detailed description.
+     *
+     * @param brokerConnectionConfig The configuration key-value map.
+     * @return Returns the created broker connection or null if there is already a connection with the same name.
+     * @throws ConfigurationException Most likely your provided name and url are invalid.
+     * @throws MqttException
+     */
+    public MqttBrokerConnection addBrokerConnection(Map<String, Object> brokerConnectionConfig)
```

Comment:
What other types of values instead of `String` do you expect here? I would suggest to expect a Map<String, String> which would result in property retrieval like

```
String brokerURL = brokerConnectionConfig.get("url");
if (StringUtils.isBlank(brokerURL) {
    throw new ConfigurationException("MQTT Broker property 'url' is not provided");
}
```

Right now passing in a blank String would satisfy all the checks but will fail on connecting to the broker.
On another note this would also make the check in line 66 obsolete.